build/speedrun_web.py

```diff
@@ -0,0 +1,510 @@
+                </div>
+            </div>
+        </div>
+    </header>
+    """
+    )
+    return
+
+
+@app.cell
+def _(mo):
+    mo.Html(
+        """
+    <div class="bg-white rounded-lg shadow p-8 mb-8">
+        <div class="flex justify-between items-start mb-6">
+            <h2 class="text-3xl font-extrabold text-gray-900 font-display">What is Speedrun?</h2>
+            <a href="https://github.com/marin-community/marin/blob/main/docs/tutorials/submitting-speedrun.md" target="_blank" rel="noopener" class="inline-flex items-center px-6 py-3 border border-transparent text-base font-medium rounded-md text-white bg-marin-blue hover:bg-blue-600 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-marin-blue transition-colors duration-150">
+                Get started with Speedrun
+            </a>
+        </div>
+        <p class="text-gray-600 leading-relaxed text-lg mb-8">
+            Speedrun is a community-driven initiative by the <a href="https://marin.community/" target="_blank" rel="noopener" class="text-marin-blue hover:text-blue-600 transition-colors duration-150">Marin project</a> to track and optimize the training efficiency of large language models. 
+            Have a new architecture or training procedure that you think is more efficient? Participate in the Marin speedrun competition (inspired by 
+            the <a href="https://github.com/KellerJordan/modded-nanogpt?tab=readme-ov-file#world-record-history" class="text-marin-blue hover:text-blue-600 transition-colors duration-150">nanogpt speedrun</a>), pick your compute budget, and create the fastest method to train a model to a 
+            certain quality! 
+        </p>
+        <p class="text-gray-600 leading-relaxed text-lg mb-8">
+            On this page, you can find leaderboards for different speedrun tracks, each targeting a specific loss threshold. You can click on any run to view the code that
+            generated it, or view the Weights & Biases link for the model! We also track the overall Pareto frontier of models, allowing us to track efficiency-performance tradeoffs across all tracks.
+        </p>
+        <p class="text-gray-600 leading-relaxed text-lg">
+            We invite you to join us in the search for more performant and efficient training methods!
+        </p>
+    </div>
+                """
+    )
+    return
+
+
+@app.cell
+def _():
+    import json
+    import pandas as pd
+
+    with open("../data/runs.json", "r") as f:
+        runs = json.load(f)
+    with open("../data/tracks.json", "r") as f:
+        tracks = json.load(f)
+    df_runs = pd.json_normalize(runs)
+    df_tracks = pd.DataFrame(tracks)
+    return df_runs, df_tracks, pd
+
+
+@app.cell
+def _(df_tracks, mo):
+    q = mo.query_params()
+    tab_map = {row["name"]: row["id"] for _, row in df_tracks.iterrows()}
+    tabs = mo.ui.tabs(
+        {row["name"]: "" for _, row in df_tracks.iterrows()},
+        value=q.get("track") or "Scaling",
```

Comment:
I think we could consider making the tracks have a stable identifier (e.g., llama_30m and a display name that we can freely edit)...so that we can maintain backward compatibility and improve the frontend over time